lib/parsers/pdf_parser/xref_table.rb

```diff
@@ -0,0 +1,99 @@
+class FormatParser::PDFParser::XRefTable
+  include FormatParser::IOUtils
+
+  XRef = Struct.new(:idx, :offset, :generation_number, :entry_type)
+
+  def self.parse(io)
+    new(io).parse
+  end
+
+  def initialize(io)
+    @io = io
+  end
+
+  # Public: parse
+  #
+  # Returns an array of XRef objects
+  def parse
+    @io.seek(@io.size - 5)
+
+    # TODO: Sometimes the EOF ends with a line-ending
+    # return [] unless safe_read(@io, 5) == FormatParser::PDFParser::EOF_MARKER
+    return [] unless xref_table_offset
+
+    @io.seek(xref_table_offset)
+
+    parse_xref_table
+  end
+
+  private
+
+  def xref_table_offset
+    @xref_table_offset ||= begin
```

Comment:
Does this value need memoization if that value only gets recovered once - and the xref table only gets parsed once per file?